src/domain/access/role-set/role.set.service.ts

```diff
@@ -756,8 +756,8 @@ export class RoleSetService {
       );
 
       if (!hasAnotherAdminRole) {
-        await this.removeContributorToImplicitRole(
-          parentRoleSet,
+        await this.removeContributorFromImplicitRole(
+          roleSet,
```

Comment:
Actually this must remain the parent role set, as that is where the "subspaceAdmin" role makes sense. 

So the logic is that if the user is no longer an admin in the subspace, and that was the last admin role, then the user is no longer a "subspace admin" in the parent space